client-common/src/key/private_key.rs

```diff
@@ -6,10 +6,81 @@ use zeroize::Zeroize;
 
 use crate::{ErrorKind, PublicKey, Result, ResultExt, SECP};
 
+/// types of private key
+#[derive(Clone)]
+pub enum PrivateKeyKind {
+    /// normal private key
+    Normal(PrivateKey),
+    // TODO: Add Hardware wallet as Privatekey
+}
+
+impl PrivateKeyTrait for PrivateKeyKind {
+    fn sign<T: AsRef<[u8]>>(&self, bytes: T) -> Result<RecoverableSignature> {
+        match self {
+            Self::Normal(pk) => pk.sign(bytes),
+        }
+    }
+    fn schnorr_sign<T: AsRef<[u8]>>(&self, bytes: T) -> Result<SchnorrSignature> {
+        match self {
+            Self::Normal(pk) => pk.schnorr_sign(bytes),
+        }
+    }
```

Comment:
this won't work with HW wallets:

1. this interface assumes it's being passed the message digest ("TXID") -- just like in https://github.com/crypto-com/chain/blob/master/client-network/src/network_ops/default_network_ops_client.rs#L317 -- for HW wallet, it won't be passed TXID, it'll be passed the *transaction data*, it'll parse this data, show some confirmation screen to the user (e.g. "confirm to sign this transaction" + details about the transaction) -- if the user presses a button on the device, inside the device, it will compute the TXID (from the transaction data e.g. sent over USB) and sign that